### ISSUE TITLE
Best way to add OOV words

### ISSUE THREAD

**Issue by chenAsaraf** [espnet/espnet#4124]
Hi,
We are using LibriSpeech recipe, with a test set that contains a lot of out of vocabulary words, which causes high WER.
We try to overcome this by adding words to the text of the language model (in train.txt), but it does not seem to help enough.

Is there any way you can suggest adding these OOV words to the data?

**Comment by sw005320** [comment]
How about using TTS (pre-trained espnet TTS or other product services) to generate such data and include them in the training data?

**Comment by chenAsaraf** [comment]
Hi again,
I am facing this 'out of vocabulary words' problem again, and I thought maybe I can change something in the LM, that would help me reduce the error around technical words/private names? 
I know that for now there is no implementation of dynamic-lexicon in ESPnet (?)
Perhaps adding OOV words to the LM training set could help? and if so, how would be the best way to do so in your opinion?
Thanks a lot, again, for this amazing tool.

**Comment by sw005320** [comment]
We have an LM shallow fusion function in general.
So, if you add OOV words to the LM training set, it can somehow mitigate the issue (but not perfectly)
